resolwe/flow/engines/__init__.py

```diff
@@ -79,25 +83,8 @@ def communicate(self, run_sync=False, verbosity=1):
                         data.save()
                         continue
 
-                    script_template = data.process.run.get('script', '')
-                    inputs = data.input.copy()
-                    hydrate_input_references(inputs, data.process.input_schema)
-                    # hydrate_input_uploads(inputs, data.process.input_schema)
-
-                    info = {}
-                    # info['case_ids'] = data.case_ids
-                    info['data_id'] = data.id
-                    # info['data_path'] = settings.RUNTIME['data_path']
-                    # info['slugs_path'] = settings.RUNTIME['slugs_path']
-                    inputs['proc'] = info  # add script info
-
-                    try:
-                        script = template.Template('{% load resource_filters %}{% load mathfilters %}' +
-                                                   script_template).render(template.Context(inputs))
-                    except template.TemplateSyntaxError as ex:
-                        data.status = Data.STATUS_ERROR
-                        data.process_error.append('Error in process script: {}'.format(ex))
-                        data.save()
+                    script = self.exprengines['dtlbash'].eval(data)
```

Comment:
I followed the concept of Django REST Framework, where you define define defaults in settings and you can override them later (i.e. in process).
@mstajdohar Where would you load expression engines from? An option is to check `exprengines` folders in all installed apps and load all `ExpressionEngine` classes. But this would be very time consuming, so it would be good to add global variable with already loaded engines (the same way as for processes in tests).